src/core/lib/gpr/sync_posix.cc

```diff
@@ -105,10 +137,29 @@ void gpr_cv_init(gpr_cv* cv) {
 #if GPR_LINUX
   GPR_ASSERT(pthread_condattr_setclock(&attr, CLOCK_MONOTONIC) == 0);
 #endif  // GPR_LINUX
+
+#ifdef GRPC_ASAN_ENABLED
+  GPR_ASSERT(pthread_cond_init(&cv->cond_var, &attr) == 0);
+  cv->leak_checker = (int*)gpr_malloc(sizeof(*cv->leak_checker));
+  GPR_ASSERT(cv->leak_checker != nullptr);
+  /* Initial it with a magic number, make no sense, just use the memory.
```

Comment:
This is similar to the above comment that I flagged. I have the same question here.

Reply:
Okay, get it.

Reply:
done